client/src/providers/AuthContext.js

```diff
@@ -1,9 +1,10 @@
-import React, { createContext, useState } from "react";
+import React, { createContext, useState, useHistory } from "react";
 import axios from "axios";
 
 export const useAuth = () => {
   const [user, setUser] = useState(null);
   const [edit, setEdit] = useState(false);
+  const history = useHistory();
```

Comment:
useHistory comes from `react-router-dom` not `react`, see the docs: https://reacttraining.com/react-router/web/api/Hooks/usehistory